cfme/web_ui/utilization.py

```diff
@@ -0,0 +1,728 @@
+            stop_dt = start_dt.replace(hour=23, minute=59, second=59)
+        elif self.option.get_interval(force_visible_text=True) == Option.IN_MOST_RECENT_HOUR:
+            bucket_duration = "1mn"
+            # sample date: 2016-10-26 07:15:48 UTC to 2016-10-26 07:30:48 UTC
+            date_range = self.option.get_range().split("to")
+            start_dt = datetime.strptime(date_range[0].strip(), "%Y-%m-%d %H:%M:%S %Z")
+            stop_dt = datetime.strptime(date_range[1].strip(), "%Y-%m-%d %H:%M:%S %Z")
+        start = self._timestamp(dt=start_dt)
+        end = self._timestamp(dt=stop_dt, end_time=True)
+        return self._list_data_mgmt(start=start, end=end, bucket_duration=bucket_duration)
+
+    def _list_data_mgmt(self, start, end, bucket_duration):
+        page = self._utz_object.page
+        mgmt = page.provider.mgmt
+        page_type = page.taggable_type
+        if page_type == "MiddlewareServer":
+            feed_id = page.feed
+            server_name = page.name
+            raw_gauge_server = partial(mgmt.metric.list_gauge_server(
+                feed_id=feed_id, server_id=server_name, bucket_duration=bucket_duration,
+                start=start, end=end))
+            raw_counter_server = partial(mgmt.metric.list_counter_server(
+                feed_id=feed_id, server_id=server_name, bucket_duration=bucket_duration,
+                start=start, end=end))
+        elif page_type == "MiddlewareDatasource":
+            feed_id = page.server.feed
+            server_name = page.server.name
+            resource_id = re.search(r'Datasource \[(.*?)\]', page.name).group(1)
+            raw_gauge_ds = partial(mgmt.metric.list_gauge_datasource(
+                feed_id=feed_id, server_id=server_name, resource_id=resource_id,
+                bucket_duration=bucket_duration, start=start, end=end))
+        elif page_type == "MiddlewareMessaging":
+            feed_id = page.server.feed
+            server_name = page.server.name
+            if page.name.startswith("JMS Topic"):
+                resource_id = re.search(r'JMS Topic \[(.*?)\]', page.name).group(1)
+                is_topic = True
+            elif page.name.startswith("JMS Queue"):
+                resource_id = re.search(r'JMS Queue \[(.*?)\]', page.name).group(1)
+                is_topic = False
+            raw_jms = partial(mgmt.metric.list_jms(
+                feed_id=feed_id, server_id=server_name, name=resource_id,
+                bucket_duration=bucket_duration, start=start, end=end))
+
+        # MiddlewareServer
+        if self.id == "jvm_heap_usage_bytes" and page_type == "MiddlewareServer":
+            raw_used = raw_gauge_server(metric_enum=MetricEnumGauge.SVR_MEM_HEAP_USED)
+            raw_max = raw_gauge_server(metric_enum=MetricEnumGauge.SVR_MEM_HEAP_MAX)
+            raw_committed = raw_gauge_server(metric_enum=MetricEnumGauge.SVR_MEM_HEAP_COMMITTED)
+            # merge all data
```

Comment:
Noted your previous response about the behavior of final_data.

If _update_mgmt_data() overwrites every time, and doesn't store a cumulative set of the data, why is it called several times here, each time assigning to final_data?  Why not just make the final call for 'committed' and return that data set?

The data flow isn't clear, and the comment of 'merge data' is misleading. Makes me think the assignment should be appending the new data set instead of overwriting final_data.